test/test_xla_sharding.py

```diff
@@ -102,6 +102,14 @@ def test_deep_copy(self):
         torch_xla._XLAC._get_xla_sharding_spec(xt),
         torch_xla._XLAC._get_xla_sharding_spec(xt2))
 
+  def test_mark_step(self):
+    xt = torch.randn(2, 4, 8, 16).to(xm.xla_device())
+    xs.mark_sharding(xt, self._get_mesh((1, 1, 1, self.n_devices)),
+                     (0, 1, 2, 3))
+    sharding_spec = torch_xla._XLAC._get_xla_sharding_spec(xt)
+    xm.mark_step()  # resets IR value
+    self.assertEqual(sharding_spec, torch_xla._XLAC._get_xla_sharding_spec(xt))
+
 
```

Comment:
can we add a test where to check after `assignirvalue`(I guess a in place operation), the sharding spec is set in the tensor IR?

Reply:
Updated the test a bit, let me know if you have any other suggestions. Thanks!